### PR TITLE
Add Cholesky and SolveTriangular for torch

### PR DESCRIPTION
**Pull request by Ch0ronomato**
## Description
Add torch cholesky operator

## Related Issue

- [ ] Closes #
- [ ] Related to #988 
- [ ] Related to #821 

## Checklist

- [X] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [X] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [X] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [X] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pytensor--1035.org.readthedocs.build/en/1035/

### CODE REVIEW

File: tests/link/pytorch/test_slinalg.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import pytensor
+from pytensor.tensor import tensor
+from pytensor.tensor.slinalg import cholesky
+
+
+torch = pytest.importorskip("torch")
+
+
+# @todo: We don't have blockwise yet for torch
```
Comment (ricardoV94) [inline tests/link/pytorch/test_slinalg.py:12]: This is a specific test, just test cholesky and solve directly with the `compare_py_and_torch` as usual

File: pytensor/link/pytorch/dispatch/slinalg.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+@pytorch_funcify.register(SolveTriangular)
+def pytorch_funcify_SolveTriangular(op, **kwargs):
+    lower = op.lower
+    trans = op.trans
+    unit_diagonal = op.unit_diagonal
+
+    def solve_triangular(A, b):
+        return torch.linalg.solve_triangular(
+            A, b, upper=not lower, unit_triangle=unit_diagonal, left=trans == "T"
```
Comment (ricardoV94) [inline pytensor/link/pytorch/dispatch/slinalg.py:25]: trans may also be `1` apparently. We should canonicalize the integers to strings so the Op only has to handle one of the encodings: https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.solve_triangular.html

**Review by ricardoV94 — COMMENTED** [review]
Title should mention SolveTriangular as well

**Comment by ricardoV94** [comment]
Actually these Ops were already being tackled in #922

**Comment by Ch0ronomato** [comment]
Gonna close this while #922 is open - can always reopen it.

**Pull request closed without merging**